partners_api/partners_api_tests/taxi_engine_tests.cpp

```diff
@@ -0,0 +1,253 @@
+#include "testing/testing.hpp"
+
+#include "partners_api/taxi_engine.hpp"
+#include "partners_api/uber_api.hpp"
+#include "partners_api/yandex_api.hpp"
+
+#include "geometry/latlon.hpp"
+
+#include "base/scope_guard.hpp"
+
+namespace
+{
+std::vector<taxi::Product> GetUberSynchronous(ms::LatLon const & from, ms::LatLon const & to)
+{
+  std::string times;
+  std::string prices;
+
+  TEST(taxi::uber::RawApi::GetEstimatedTime(from, times), ());
+  TEST(taxi::uber::RawApi::GetEstimatedPrice(from, to, prices), ());
+
+  size_t reqId = 0;
+  taxi::uber::ProductMaker maker;
+  std::vector<taxi::Product> uberProducts;
+
+  maker.Reset(reqId);
+  maker.SetTimes(reqId, times);
+  maker.SetPrices(reqId, prices);
+  maker.MakeProducts(
+      reqId, [&uberProducts](vector<taxi::Product> const & products) { uberProducts = products; },
+      [](taxi::ErrorCode const code) { TEST(false, ()); });
+
+  return uberProducts;
+}
+
+std::vector<taxi::Product> GetYandexSynchronous(ms::LatLon const & from, ms::LatLon const & to)
+{
+  std::string yandexAnswer;
+  std::vector<taxi::Product> yandexProducts;
+
+  TEST(taxi::yandex::RawApi::GetTaxiInfo(from, to, yandexAnswer), ());
+
+  taxi::yandex::MakeFromJson(yandexAnswer, yandexProducts);
+
+  return yandexProducts;
+}
+
+taxi::ProvidersContainer GetProvidersSynchronous(ms::LatLon const & from, ms::LatLon const & to)
+{
+  taxi::ProvidersContainer providers;
+
+  providers.emplace_back(taxi::Provider::Type::Uber, GetUberSynchronous(from, to));
+  providers.emplace_back(taxi::Provider::Type::Yandex, GetYandexSynchronous(from, to));
+
+  return providers;
+}
+
+void CompareProviders(taxi::ProvidersContainer const & providersContainer,
+                      taxi::ProvidersContainer const & synchronousProviders)
+{
+  TEST_EQUAL(synchronousProviders.size(), providersContainer.size(), ());
+
+  for (auto const & sp : synchronousProviders)
```

Comment:
Am I right that you're checking here that there's a one-to-one correspondence between items in synchronousProviders and providersContainer?